File: 1_Review/javascript/4_custom_modules_and_classes.js
// 4) Custom classes (defined in this file)
// Run: node 4_custom_modules_and_classes.js

// Classes are usually abstractions of real-world objects and processes.
class Person {
  constructor(name, age) {
    this.name = name;
    this.age = age;
  }

  // Instance methods use `this`
  greet() {
    console.log(`Hello, my name is ${this.name} and I am ${this.age} years old.`);
  }
}

const p1 = new Person('Alice', 30);
p1.greet();

const p2 = new Person('Bob', 25);
p2.greet();

File: 1_Review/python/4_custom_modules_and_classes.py
# defining the Person class
# classes are typically abstractions of real world objects and processes
class Person:
    def __init__(self, name, age):
        # defining instance variables  (properties of the object.  e.g. properties of a person)
        # must prefix properties with self when reading and writing to them
        self.name = name
        self.age = age

    # all instance methods should have 'self' as the first parameter
    def greet(self):
        print(f"Hello, my name is {self.name} and I am {self.age} years old.")



# first instance of Person (self in class will point to this instance)
p1 = Person("Alice", 30)
p1.greet()

# second instance of Person (self in class will point to this instance)
p2 = Person("Bob", 25)
p2.greet()
